### ISSUE TITLE
Variable Interpolation

### ISSUE THREAD

**Issue by Pax-Newman** [sizemore0125/skeletonkey#34]
I'd like to propose adding variable interpolation like in [OmegaConf](https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#variable-interpolation). Nested interpolation and custom resolvers would be a really nice bonus but basic interpolation would be the most helpful.

**Comment by Diego-Llanes** [comment]
This would be awesome

**Comment by sizemore0125** [comment]
I agree. I think this feature would really expand the use-cases of skeletonkey.

**Comment by sizemore0125** [comment]
This was added in this commit: f2f5d67a2837818d901434721dbcdcfb012d1f2c

There may need more work in how we restrict what should or should not be interpolated. However, a simple form of this now exists.

```yaml
model:
  _instance_: main.MyModel
  activation: relu
  debug: False

dataloader:
  _instance_: main.MyDataLoader
  batch_size: 32
  debug: ${model.debug}
```
```python
print(args.dataloader.debug) # False
```
